src/tests/functional/link_prefetch_observer_tests.js

```diff
@@ -62,6 +62,22 @@ test("it doesn't prefetch the page when link has data-turbo=false", async ({ pag
   await assertNotPrefetchedOnHover({ page, selector: "#anchor_with_turbo_false" })
 })
 
+test("allows to cancel prefetch requests with custom logic", async ({ page }) => {
+  await goTo({ page, path: "/hover_to_prefetch.html" })
+
+  await assertPrefetchedOnHover({ page, selector: "#anchor_with_remote_true" })
+
+  await page.evaluate(() => {
+    document.body.addEventListener("turbo:before-prefetch", (event) => {
+      if (event.target.hasAttribute("data-remote")) {
+        event.preventDefault()
+      }
+    })
+  })
+
+  await assertNotPrefetchedOnHover({ page, selector: "#anchor_with_remote_true" })
+})
+
```

Comment:
We might want to add a test for not prefetching when `data-turbo-prefetch` is set to `"false"`.

Reply:
That seems already covered in [its own test](https://github.com/hotwired/turbo/blob/7327a955ea93ef885ea157977acaeb96068f80ba/src/tests/functional/link_prefetch_observer_tests.js#L55-L58). The features are fairly independent, so no need to test their interaction.